contracts/truefi2/PoolFactory.sol

```diff
@@ -0,0 +1,93 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.6.10;
+
+import {Ownable} from "../common/UpgradeableOwnable.sol";
+import {OwnedProxyWithReference} from "../proxy/OwnedProxyWithReference.sol";
+import {ERC20} from "../common/UpgradeableERC20.sol";
+
+import {ITrueFiPool2} from "./interface/ITrueFiPool2.sol";
+import {ImplementationReference} from "../proxy/ImplementationReference.sol";
+
+contract PoolFactory is Ownable {
+    // ================ WARNING ==================
+    // ===== THIS CONTRACT IS INITIALIZABLE ======
+    // === STORAGE VARIABLES ARE DECLARED BELOW ==
+    // REMOVAL OR REORDER OF VARIABLES WILL RESULT
+    // ========= IN STORAGE CORRUPTION ===========
+
+    // @dev Mapping of ERC20 token's addresses to its pool's addresses
+    mapping(address => address) public pool;
+    mapping(address => bool) public isPool;
+
+    // @dev Whitelist for tokens, which can have pools created
+    mapping(address => bool) public isAllowed;
+
```

Comment:
add a bool enforceAllowed which is set to true when initializing the contract.
also add a function to set enforceAllowed that is onlyOwner.
this allows us to disable the whitelist after some period.